catalog/transformer/catalog-transformer-pptx/src/main/java/ddf/catalog/transformer/input/pptx/PptxInputTransformer.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.input.pptx;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.pdfbox.tools.imageio.ImageIOUtil;
+import org.apache.poi.sl.usermodel.SlideShow;
+import org.apache.poi.sl.usermodel.SlideShowFactory;
+import org.apache.poi.xslf.usermodel.XMLSlideShow;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.FileBackedOutputStream;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+
+/**
+ * This is a decorator class that adds a thumbnail for PPTX files. It relies on
+ * an injected {@link InputTransformer} to extract the metadata, and then it generates a
+ * thumbnail image with Apache POI.
+ */
+public class PptxInputTransformer implements InputTransformer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PptxInputTransformer.class);
+
+    private static final int RESOLUTION_DPI = 44;
+
+    private static final float IMAGE_QUALITY = 1.0f;
+
+    private static final float IMAGE_HEIGHTWIDTH = 128;
+
+    private static final String FORMAT_NAME = "jpg";
+
+    private static final int FILE_BACKED_THRESHOLD = 1000000;
+
+    private final InputTransformer inputTransformer;
+
+    /**
+     * The inputTransformer parameter will be used to generate the basic metadata. If the parameter
+     * is null, then a {@link NullPointerException} will be thrown.
+     *
+     * @param inputTransformer must be non-null
+     * @throws NullPointerException
+     */
+    public PptxInputTransformer(InputTransformer inputTransformer) {
+
+        if (inputTransformer == null) {
+            throw new NullPointerException("The inputTransformer parameter must be non-null");
+        }
```

Comment:
These three lines of code can be reduced to only one: 
`notNull(inputTransformer, "The input transformer cannot be null");`
You'll have to add the `@NotNull` annotation in front of the parameter declaration like so: 
`public PptxInputTransformer(@NotNull InputTransformer inputTransformer) {`